doc/concepts/configuration.rst

```diff
@@ -171,55 +178,128 @@ You can learn more about configuring replication from :ref:`Replication tutorial
+The example below shows how to enable the ``roles.crud-router`` role provided by the `CRUD <https://github.com/tarantool/crud>`__ module using the :ref:`roles <configuration_reference_roles>` option:
+
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster_crud/config.yaml
     :language: yaml
+    :start-at: roles.crud-router
+    :end-at: roles.crud-router
     :dedent:
 
-To get a value of the custom ``greeting`` property in the application code,
-use the ``config:get()`` function provided by the :ref:`config <config-module>` module.
+Similarly, you can enable the ``roles.crud-storage`` role to make instances act as CRUD storages:
 
-..  literalinclude:: /code_snippets/snippets/config/instances.enabled/application/myapp.lua
-    :language: lua
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster_crud/config.yaml
+    :language: yaml
+    :start-at: roles.crud-storage
+    :end-at: roles.crud-storage
     :dedent:
 
-As a result of :ref:`starting <configuration_run_instance>` the *instance001*, a log should contain the following line:
+Example on GitHub: `sharded_cluster_crud <https://github.com/tarantool/doc/tree/latest/doc/code_snippets/snippets/sharding/instances.enabled/sharded_cluster_crud>`_
 
-.. code-block:: console
 
-    main/103/interactive/myapp I> Hello from app, instance001!
+.. _configuration_application_roles_configure:
 
-The ``app`` section can be placed in any :ref:`configuration scope <configuration_scopes>`.
-As an example use case, you can provide different applications for storages and routers in a sharded cluster:
+Configuring a role
+******************
 
-.. code-block:: yaml
+The :ref:`roles_cfg <configuration_reference_roles_cfg>` option allows you to specify the configuration for each role.
+In this option, the role name is the key and the role configuration is the value.
+
+The example below shows how to enable statistics on called operations by providing the ``roles.crud-router`` role's configuration:
+
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster_crud/config.yaml
+    :language: yaml
+    :start-at: roles.crud-router
+    :end-at: stats_quantile_max_age_time
+    :dedent:
+
+Example on GitHub: `sharded_cluster_crud <https://github.com/tarantool/doc/tree/latest/doc/code_snippets/snippets/sharding/instances.enabled/sharded_cluster_crud>`_
+
+
+
+.. _configuration_application_roles_scopes:
+
+Roles and configuration scopes
+******************************
+
+As the most of configuration options, roles and their configurations can be defined at :ref:`different levels <configuration_scopes>`.
+Given that the ``roles`` option has the ``array`` type and ``roles_cfg`` has the ``map`` type, there are some specifics of applying the configuration:
+
+-   For ``roles``, an instance's role takes precedence over roles defined at another level.
```

Comment:
```suggestion
-   For ``roles``, an instance's role overwrites all roles defined at another levels.
```
Nit: `overwrites` seems a more straightforward verb to explain that other roles are discarded.